File: Serene/Serene.Script/Northwind/Customer/CustomerEditor.cs

namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;

    public class CustomerEditor : LookupEditorBase<CustomerRow>
    {
        public CustomerEditor(jQueryObject container, LookupEditorOptions options)
            : base(container, options)
        {
        }

        protected override string GetLookupKey()
        {
            return CustomerRow.LookupKey;
        }

        protected override string GetItemText(CustomerRow item, Lookup<CustomerRow> lookup)
        {
            return base.GetItemText(item, lookup) + " [" + item.CustomerID + "]";
        }
    }
}

File: Serene/Serene.Web/Modules/Northwind/Customer/CustomerEditor.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerEditor()
    export class CustomerEditor extends Serenity.LookupEditorBase<Serenity.LookupEditorOptions, CustomerRow> {

        constructor(hidden: JQuery) {
            super(hidden);
        }

        protected getLookupKey() {
            return CustomerRow.lookupKey;
        }

        protected getItemText(item, lookup) {
            return super.getItemText(item, lookup) + ' [' + item.CustomerID + ']';
        }
    }
}